scroll-animations-1/Overview.bs

```diff
@@ -789,28 +770,66 @@ follows:
+1.  Let <var>first offset</var> be true.
+1.  If {{ScrollTimeline/scrollOffsets}} is empty
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+1.  If {{ScrollTimeline/scrollOffsets}} has one element
+    1.  Run the procedure to [=resolve a scroll timeline offset=] for 'auto'
+        with the <var>is first</var> flag set to <var>first offset</var> and
+        add the resulted value into <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  For each <var>scroll offset</var> in the list of
+    {{ScrollTimeline/scrollOffsets}}, perform the following steps:
+    1.  Let <var>effective offset</var> be a result of applying the procedure
+        to [=resolve a scroll timeline offset=] for <var>scroll offset</var>
+        with the <var>is first</var> flag set to <var>first offset</var>.
+    1.  If <var>effective offset</var> is null, the
+        [=effective scroll offsets=] is empty and abort the remaining steps.
+    1.  Add <var>effective offset</var> into
+        <var>effective scroll offsets</var>.
+    1.  Set <var>first offset</var> to false.
+1.  Return <var>effective scroll offsets</var>.
+
+### The progress of a {{ScrollTimeline}} ### {#progress-calculation-algorithm}
+The <dfn>scroll timeline progress</dfn> is calculated by applying the procedure
+to <dfn>calculate scroll timeline progress</dfn> given |offset| as follows:
+
+1.  Let <var>scroll offsets</var> be a result of applying the procedure to
+    [=resolve scroll timeline offsets=] for {{scrollOffsets}}.
+1.  Let <var>offset id</var> correspond to the position of the last offset in
+    <var>scroll offsets</var> whose value is less than or equal to
+    <var>offset</var> and the value at the following position greater than
+    <var>offset</var.
+1.  Let <var>start offset</var> be the offset value at position
+    <var>offset id</var> in <var>scroll offsets</var>.
+1.  Let <var>end offset</var> be the value of next offset in <var>scroll
+    offsets</var> after <var>start offset</var>.
+1.  Let <var>size</var> be the number of offsets in <var>scroll offsets</var>.
+1.  Let <var>offset weight</var> be the result of evaluating
+    <code> 1 / (<var>size</var> - 1)</code>.
+1.  Let <var>interval progress</var> be the result of evaluating
+    <code>(<var>offset</var> - <var>start offset</var>) /
+    (<var>end offset</var> - <var>start offset</var>)</code>.
+1.  Return the result of evaluating
+    <code>(<var>offset id</var> - 1 + <var>interval progress</var>) &times;
```

Comment:
> `<var>offset id</var> - 1`

I guess this means that `offset id` starts at `1`? Can't we just specify earlier that it starts at zero?

Reply:
Thanks for pointing this out. It's a copy and paste from the [implementation](https://source.chromium.org/chromium/chromium/src/+/master:cc/animation/scroll_timeline.h;l=149;drc=131697ea2c25234744fe1917571bfa454c2d22cb?q=scroll_timeline&ss=chromium) where offset_id correspond to the end of the matching interval.
I added a comment to [crbug](https://bugs.chromium.org/p/chromium/issues/detail?id=1094014#c7) to update the implementation based on spec.